src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -0,0 +1,179 @@
+
+    UserTerm term = new UserTerm();
+
+    term.setUserId(user.getUserId());
+
+    return term;
+  }
+
+  public StudyTerm toStudyTerm(Study study) {
+    StudyTerm term = new StudyTerm();
+
+    term.setDescription(study.getDescription());
+    term.setStudyName(study.getName());
+    term.setStudyId(study.getStudyId());
+    term.setDataTypes(study.getDataTypes());
+    term.setPiName(study.getPiName());
+    term.setPublicVisibility(study.getPublicVisibility());
+
+    findStudyProperty(
+        study.getProperties(), "phenotypeIndication"
+    ).ifPresent(
+        prop -> term.setPhenotype((String) prop.getValue())
+    );
+
+    findStudyProperty(
+        study.getProperties(), "species"
+    ).ifPresent(
+        prop -> term.setSpecies((String) prop.getValue())
+    );
+
+    findStudyProperty(
+        study.getProperties(), "dataCustodianEmail"
+    ).ifPresent(
+        prop -> term.setDataCustodian((String) prop.getValue())
+    );
+
+    User dataSubmitter = userDAO.findUserById(study.getCreateUserId());
+    term.setDataSubmitter(toUserTerm(dataSubmitter));
+
+    return term;
+  }
+
+  public DatasetTerm toDatasetTerm(Dataset dataset) {
+    DatasetTerm term = new DatasetTerm();
+
+    term.setDatasetId(dataset.getDataSetId());
+    term.setDatasetIdentifier(dataset.getDatasetIdentifier());
+
+    if (Objects.nonNull(dataset.getStudy())) {
+      term.setStudy(toStudyTerm(dataset.getStudy()));
+    }
+
+    term.setDacId(dataset.getDacId());
+
+    Collection<Integer> approvedUserIds =
+        dataAccessRequestDAO.findAllUserIdsWithApprovedDARsByDatasetId(
+            dataset.getDataSetId());
+
+    if (!approvedUserIds.isEmpty()) {
+      Collection<User> approvedUsers = userDAO.findUsers(approvedUserIds);
+
+      List<UserTerm> approvedUserTerms = approvedUsers.stream().map(this::toUserTerm).toList();
+      term.setApprovedUsers(approvedUserTerms);
+    }
+
+    term.setDataUse(useRestrictionConverter.translateDataUseSummary(dataset.getDataUse()));
+
+    findDatasetProperty(
+        dataset.getProperties(), "openAccess"
+    ).ifPresent(
+        datasetProperty -> term.setOpenAccess((Boolean) datasetProperty.getPropertyValue())
+    );
+
+    // TODO: unsure how to get number of participants; there could
+    // be multiple if there are multiple file types
+
+    findDatasetProperty(
+        dataset.getProperties(), "url"
+    ).ifPresent(
+        datasetProperty -> term.setUrl((String) datasetProperty.getPropertyValue())
+    );
+
+    findDatasetProperty(
+        dataset.getProperties(), "dataLocation"
+    ).ifPresent(
+        datasetProperty -> term.setDataLocation((String) datasetProperty.getPropertyValue())
+    );
+
+    return term;
+  }
+
+  Optional<DatasetProperty> findDatasetProperty(Collection<DatasetProperty> props,
+      String schemaProp) {
+    return
+        props
+            .stream()
+            .filter((p) -> p.getSchemaProperty().equals(schemaProp))
```

Comment:
`p.getSchemaProperty()` can be null ...
```suggestion
            .filter(Objects::nonNull)
            .filter(p -> p.getSchemaProperty().equals(schemaProp))
```